lib/src/key_pair/key_pair.ts

```diff
@@ -65,4 +65,16 @@ export default class KeyPair {
     public get privateKey(): Buffer | undefined {
         return this.innerPrivateKey;
     }
+
+    public toObject(): NativeKeyPair {
```

Comment:
Cont. with previous confusion, if the `KeyPair` class here comes with keys of `publicKey` and `privateKey`. Then `toObject` could be implemented with following:
```
return Object.assign({}, this);
```
Which could be more maintainable (keep the consistency of the keys).

Reply:
`Object.assign({}, this);` is also an interesting solution. 

My intention is:
1. Expose `publicKey` and `privateKey` getter such that it is not modifiable*
2. `toObject` returns an object with the core props of `KeyPair` class: `publicKey` and `privateKey`

If I use `Object.assign({}, this);`, it would expose possibly any private key defined in the future. But it does achieve the immutability of props.

*However, I later realize `getter` still allows it to be modified. Maybe what I need is to clone on return or freeze the value in the first place. What do you think?

Reply:
>  If I use Object.assign({}, this);, it would expose possibly any private key defined in the future.
From the names of the class and method `keyPair.toObject()`, it should expose `keyPair`'s all keys to the user. So I vote for `Object.assign({}, this);`

Btw, I think your getter now is fine for the immutability of props.